### ISSUE TITLE
lint needless borrows in format args

### ISSUE THREAD

**Issue by matthiaskrgr** [rust-lang/rust-clippy#7961]
This could probably be its own lint rather than an extension of `needless_borrow`:

When passing args to `format!()` and and friends, we actually don't need to explicitly use a `&borrow` because we don't consume the object:
````rust
    println!("{:?} {} {}", &a, &b, &c.display());
```` 
could as well be:
````rust
    println!("{:?} {} {}", a, b, c.display());
```` 

as this code sample demonstrates:
````rust
pub fn main() {
    let a = vec!['a'];
    let b = "hello";
    let c = std::path::PathBuf::from("/");

    println!("{:?} {} {}", a, b, c.display());

    std::mem::drop(a);
    std::mem::drop(b);
    std::mem::drop(c);
}
````
